src/slash/deck.ts

```diff
@@ -80,9 +81,134 @@ export class DeckCommand extends AutocompletableCommand {
+	const row = new ActionRowBuilder<ButtonBuilder>().addComponents(
+		new ButtonBuilder()
+			.setCustomId(encodeCustomId("accept", tournament.tournamentId))
+			.setLabel("Accept")
+			.setStyle(ButtonStyle.Success),
+		new ButtonBuilder()
+			.setCustomId(encodeCustomId("reject", tournament.tournamentId))
+			.setLabel("Reject")
+			.setStyle(ButtonStyle.Danger)
+	);
+	return row;
+}
+
+export class AcceptButtonHandler implements ButtonClickHandler {
+	readonly buttonIds = ["accept"];
+
+	async click(interaction: ButtonInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const modal = new ModalBuilder()
+			.setCustomId(encodeCustomId("acceptModal", tournamentIdString))
+			.setTitle("Accept Deck");
+		const deckLabelInput = new TextInputBuilder()
+			.setCustomId("acceptDeckLabel")
+			.setLabel("What is the deck's theme?")
+			.setStyle(TextInputStyle.Short)
+			.setRequired(false);
+		const actionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(deckLabelInput);
+		modal.addComponents(actionRow);
+		await interaction.showModal(modal);
+	}
+}
+
+export class RejectButtonHandler implements ButtonClickHandler {
+	readonly buttonIds = ["reject"];
+
+	async click(interaction: ButtonInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const modal = new ModalBuilder()
+			.setCustomId(encodeCustomId("rejectModal", tournamentIdString))
+			.setTitle("Reject Deck");
+		const rejectReasonInput = new TextInputBuilder()
+			.setCustomId("rejectReason")
+			.setLabel("Why is the deck illegal?")
+			.setStyle(TextInputStyle.Paragraph)
+			.setRequired(false);
+		const actionRow = new ActionRowBuilder<TextInputBuilder>().addComponents(rejectReasonInput);
+		modal.addComponents(actionRow);
+		await interaction.showModal(modal);
+	}
+}
+
+export class AcceptLabelModal implements MessageModalSubmitHandler {
+	readonly modalIds = ["acceptModal"];
+
+	async submit(interaction: ModalMessageModalSubmitInteraction, ...args: string[]): Promise<void> {
+		const tournamentIdString = args[0];
+		const tournamentId = parseInt(tournamentIdString, 10);
+		const deck = await ManualDeckSubmission.findOneOrFail({
+			where: {
+				discordId: interaction.user.id,
+				tournamentId: tournamentId
+			},
+			relations: ["tournament"]
+		});
+		const tournament = deck.tournament;
+		const player = await interaction.client.users.fetch(deck.discordId);
+
+		let label: string | undefined = interaction.fields.getTextInputValue("acceptDeckLabel");
+		if (label.length === 0) {
+			label = undefined;
+		}
+
+		// set deck as approved and add label
+		// manual query to workaround bug
+		await ManualDeckSubmission.createQueryBuilder()
+			.update()
+			.set({ approved: true, label })
+			.where("tournamentId = :tournamentId AND discordId = :discordId", {
+				tournamentId,
+				discordId: interaction.user.id
+			})
+			.execute();
```

Comment:
Check the response on this. If the number of updated rows is 0, then we didn't update anything (which can probably only happen in some kind of race condition?)

Reply:
result.affected is possibly undefined because not all database engines support it. Can we confirm that postgres and sqlite do support it and we can asset not-null? Or is this an issue we have to think more about.

Reply:
I can check now but I almost guarantee you that it is. When running a query in psql, I can see the number of rows affected.